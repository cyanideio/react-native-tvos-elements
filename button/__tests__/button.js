import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TvOSButton from '../../button';

describe('TvOSButton Component', () => {
    it('should render without issues', () => {
        const component = shallow(<TvOSButton title='bitcoin' />);
        expect(component.length).toBe(1);
    });
});
