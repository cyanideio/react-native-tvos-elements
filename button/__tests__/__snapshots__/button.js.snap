// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvOSButton Component should render: <custom text style> 1`] = `
<View>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onHideUnderlay={[Function]}
    onPress={[Function]}
    onShowUnderlay={[Function]}
    tvParallaxProperties={
      Object {
        "magnification": 1.05,
        "pressDuration": 0.3,
        "shiftDistanceX": 0,
        "shiftDistanceY": 2,
        "tiltAngle": 0,
      }
    }
    underlayColor="transparent"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 5,
            "height": 74.11111111111111,
            "justifyContent": "center",
            "marginBottom": 15,
            "width": 250,
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "backgroundColor": "rgba(255,255,255,0.1)",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 30,
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "color": "white",
            },
            Object {
              "color": "black",
            },
          ]
        }
      >
        Settings
      </Text>
      <Icon />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`TvOSButton Component should render: <icon left> 1`] = `
<View>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onHideUnderlay={[Function]}
    onPress={[Function]}
    onShowUnderlay={[Function]}
    tvParallaxProperties={
      Object {
        "magnification": 1.05,
        "pressDuration": 0.3,
        "shiftDistanceX": 0,
        "shiftDistanceY": 2,
        "tiltAngle": 0,
      }
    }
    underlayColor="transparent"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 5,
            "height": 74.11111111111111,
            "justifyContent": "center",
            "marginBottom": 15,
            "width": 250,
          },
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "backgroundColor": "rgba(255,255,255,0.1)",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 30,
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "color": "white",
            },
            undefined,
          ]
        }
      >
        Steam
      </Text>
      <Icon
        iconName="ios-person"
        iconType="Ionicons"
      />
    </View>
  </TouchableHighlight>
</View>
`;

exports[`TvOSButton Component should render: <title only> 1`] = `
<View>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onHideUnderlay={[Function]}
    onPress={[Function]}
    onShowUnderlay={[Function]}
    tvParallaxProperties={
      Object {
        "magnification": 1.05,
        "pressDuration": 0.3,
        "shiftDistanceX": 0,
        "shiftDistanceY": 2,
        "tiltAngle": 0,
      }
    }
    underlayColor="transparent"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 5,
            "height": 74.11111111111111,
            "justifyContent": "center",
            "marginBottom": 15,
            "width": 250,
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "backgroundColor": "rgba(255,255,255,0.1)",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 30,
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "color": "white",
            },
            undefined,
          ]
        }
      >
        bitcoin
      </Text>
      <Icon />
    </View>
  </TouchableHighlight>
</View>
`;
